lte/gateway/python/magma/pipelined/ebpf/ebpf_manager.py

```diff
@@ -306,22 +309,23 @@ def _get_if_name(self, if_index: int):
         return None
 
     def _get_mac_address(self, if_name: str):
-        addr_str = netifaces.ifaddresses(self.sgi_if_name)[netifaces.AF_LINK][0]['addr']
+        addr_str = get_mac_address_from_iface(if_name)
         LOG.debug("if-name: %s, mac: %s" % (if_name, addr_str))
         return self._pack_mac_addr(addr_str)
 
     def _get_mac_address_of_ip(self, ip_addr: IPAddress):
-        if ip_addr.version == IPAddress.IPV4:
-            ip_str = socket.inet_ntop(socket.AF_INET, ip_addr.address)
-            addr_str = getmacbyip(ip_str)
-        else:
-            ip_str = socket.inet_ntop(socket.AF_INET6, ip_addr.address)
-            addr_str = getmacbyip6(ip_str)
-        if not addr_str:
-            LOG.error("Coudn't find mac for IP: %s, disabling ebpf" % (ip_str))
+        ip_str = ""
+        try:
+            if ip_addr.version == IPAddress.IPV4:
+                ip_str = socket.inet_ntop(socket.AF_INET, ip_addr.address)
+                addr_str = get_mac_address_from_ip4(ip_str)
```

Comment:
@sebathomas These gateway IPs are remote ips. 

We could use the mac addresses coming from mobilityd:
https://github.com/magma/magma/blob/a5d0ba40af4db5c9130cfcf7ea86bb9c4840c2f4/lte/protos/mobilityd.proto#L133

But these are potentially wrong. I would suggest to leave this as is for now and open up a follow up ticket for this case.

Reply:
For IP4, this is what `_get_gw_mac_address_v4` is doing. For IPv6, we should probably change back to scapy as well in `_get_gw_mac_address_v6` and add that to the new issue.

Reply:
Yepp v4 looks fine maybe you want to mention this in #14442. ipv6 does not use arp but NDP (neigh).